src/components/pages/personal-teacher-page/personal-teacher-card/PersonalTeacherCard.tsx

```diff
@@ -15,9 +16,24 @@ export type PersonalTeacherCardProps = GetTeacherDTO;
 
 const PersonalTeacherCard: FC<PersonalTeacherCardProps> = props => {
   const [isContactsVisible, setContactsVisibility] = useState(false);
-
+  const blockRef = useRef(null);
+  const { setFloatingCardShowed } = useContext(TeacherContext);
+  useEffect(() => {
+    const handleScroll = () => {
+      const bottom = blockRef.current?.getBoundingClientRect().bottom;
+      if (bottom < 0) {
+        setFloatingCardShowed(true);
+      } else {
+        setFloatingCardShowed(false);
+      }
+    };
+    window.addEventListener('scroll', handleScroll);
+    return () => {
+      window.removeEventListener('scroll', handleScroll);
+    };
```

Comment:
why do you need to return this also?

Reply:
it's stackoverflow code, I'm afraid to change it) and does it make sense?